### ISSUE TITLE
Exceptions in code called from GOAL cause crashes on Windows

### ISSUE THREAD

**Issue by chillypepper** [open-goal/jak-project#2025]
### Acknowledgements

- [X] I have [searched](https://github.com/open-goal/jak-project/issues?q=is%3Aissue+is%3Aopen+label%3Abug+sort%3Aupdated-desc) for duplicate or closed bug reports
- [X] I understand that I am supposed to provide my own legitimately obtained copy of the game

### Describe the Bug

If any exception is thrown inside a C++ function called from goal the program crashes, even if the exception is caught and handled properly. A quick example to replicate:

```
u64 CPadGetData(u64 cpad_info) {
  try {
    throw 1;
  } catch (...) {
    lg::debug("TEST INSIDE CPadGetData");
  }
  ...
```

If you modify `CPadGetData` in `game\kernel\common\kmachine.cpp` you'll get a crash quickly after start-up. If you have the same try/catch set-up in functions not called by goal though (one quick example being `ee_runner` in `game\runtime.cpp` then it works exactly as expected, with a log printed and no crashing.

-----

I originally noted this in the discord development channel (link to message: https://discord.com/channels/756287461377703987/756287461830688810/1043410127778234408) and I'm raising this issue with information from those messages while I investigate.

This seems to be a Windows only problem, and only when C++ code is called from within goal. Water has suggested that it might be an issue with Windows exceptions not playing nicely with the way GOAL sets up the stack (visible in discord from above link).

My next steps are to see if I can get a nice view of what's actually being generated for these throw/catch statements on Windows to assist with debugging/understanding the problem, and separately to understand more about how GOAL is calling C++ code to find problem areas (as well as how the GOAL stack is set up initially)

### How To Reproduce

A quick example to replicate:

```
u64 CPadGetData(u64 cpad_info) {
  try {
    throw 1;
  } catch (...) {
    lg::debug("TEST INSIDE CPadGetData");
  }
  ...
```

If you modify `CPadGetData` in `game\kernel\common\kmachine.cpp` you'll get a crash quickly after start-up. If you have the same try/catch set-up in functions not called by goal though (one quick example being `ee_runner` in `game\runtime.cpp` then it works exactly as expected, with a log printed and no crashing.

### Does this problem occur on original hardware or PCSX2?

Not needed, bug is obvious

### Expected Behavior

Handled exceptions should continue smoothly through to the catch statements and not crash

### Operating System

Windows 10 Pro 22H2 - 19045.2251 - Likely affects a large number of Windows installs

### OpenGOAL Version

029983270

### Game Version

PAL

### Are you using a high refresh-rate monitor (>60Hz)

Yes

**Comment by xTVaser** [comment]
TLDR - https://discord.com/channels/756287461377703987/756287461830688810/1043670898932195428

Using https://github.com/boostorg/coroutine either in full or in part has come up both times this issue has come up.
